azmq/detail/config/thread.hpp

```diff
@@ -0,0 +1,26 @@
+/*
+    Copyright (c) 2013-2014 Contributors as noted in the AUTHORS file
+
+    This file is part of azmq
+
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+*/
+#ifndef AZMQ_DETAIL_CONFIG_THREAD_HPP_
+#define AZMQ_DETAIL_CONFIG_THREAD_HPP_
+
+#if !defined(AZMQ_HAS_STD_THREAD)
```

Comment:
OK, now I understand.

Then I propose the next approach.

Support two main scenarios:
* AZMQ using Boost ASIO and Boost config macros and probably some other Boost stuff required by ASIO (like system error) but all the rest is from std (default case)
* Completely Boost free AZMQ

For each scenario it should be possible to "backup" some std features to Boost, for example, thread-related stuff (like thread, mutex, conditional_variable, unique_lock) and regexes.

For main scenario I propose a define `AZMQ_USE_BOOST_ASIO`. For backing up features I propose defines `AZMQ_DISABLE_STD_THREAD` and `AZMQ_DISABLE_STD_REGEX`.

Btw, there is an information about std/boost dependency of Asio/Boost.Asio: http://think-async.com/Asio/AsioAndBoostAsio